misc/helm-charts/publish.sh

```diff
@@ -97,6 +97,72 @@ else
   exit 0
 fi
 
+echo "--- Verifying that Helm Chart has been published"
+i=0
+HELM_CHART_PUBLISHED=false
+while (( i < 30 )); do
+  YAML=$(curl -s "https://materializeinc.github.io/materialize/index.yaml")
+  CURRENT_HELM_CHART_VERSION=$(echo "$YAML" | yq '.entries["materialize-operator"][0].version')
+  CURRENT_MZ_VERSION=$(echo "$YAML" | yq '.entries["materialize-operator"][0].appVersion')
+  if [[ "$CURRENT_HELM_CHART_VERSION" == "$CI_HELM_CHART_VERSION" && "$CURRENT_MZ_VERSION" == "$CI_MZ_VERSION" ]]; then
+    echo "Helm Chart $CURRENT_HELM_CHART_VERSION with Materialize $CURRENT_MZ_VERSION has successfully been published"
+    HELM_CHART_PUBLISHED=true
+    break
+  fi
+  echo "Latest version seems to be Helm Chart $CURRENT_HELM_CHART_VERSION with Materialize $CURRENT_MZ_VERSION"
+  sleep 10
+  ((i += 1))
+done
+
+if [ "$HELM_CHART_PUBLISHED" = false ]; then
+  echo "Failing because Helm Chart was not successfully published"
+  exit 1
+fi
+
+echo "--- Bumping terraform-helm-materialize"
+rm -rf terraform-helm-materialize
+git clone https://"$GITHUB_TOKEN"@github.com/MaterializeInc/terraform-helm-materialize.git
+cd terraform-helm-materialize
+sed -i "s/\".*\"\(.*\) # META: helm-chart version/\"$CI_HELM_CHART_VERSION\"\\1 # META: helm-chart version/" variables.tf
+sed -i "s/\".*\"\(.*\) # META: mz version/\"$CI_MZ_VERSION\"\\1 # META: mz version/" variables.tf
```

Comment:
I couldn't find any tool that can edit our variables.tf files cleanly, tried a bunch. Thus the META comments to make sed work.

Reply:
huh... so this is a good idea, but it probably breaks when we have two LTS versions and we need to patch the prior one. I mean it won't break break, but the newest TF module may point to the newest version of the prior LTS, which would be confusing. 

We might need to use the same name for the helm chart and LTS versions if we're going to upgrade them in lockstep?

Reply:
I think we can check if there are higher versions released already (for example v25.2.5) and then not mark the newly released v25.1.13 as the latest version etc.

Reply:
This is addressed in https://github.com/MaterializeInc/materialize/pull/32431/commits/2009dfb9a55bc930fad0751c4eb931641a247d0d